src/api/app/policies/comment_policy.rb

```diff
@@ -32,10 +32,10 @@ def reply?
     !(user.blank? || user.is_nobody? || record.user.is_nobody?)
   end
 
-  # Only logged-in Admins of Staff members can moderate comments
+  # Only logged-in Admins/Staff members or user with moderator role can moderate comments
```

Comment:
I would also not show the `Moderate` button in case the user is the author

`return false if user == record.user`

in that case the user should edit/delete its own comment.

Reply:
nice , thank you